pkg/api/v1/defaults.go

```diff
@@ -160,6 +160,18 @@ func SetDefaults_Pod(obj *Pod) {
 			}
 		}
 	}
+	for i := range obj.Spec.InitContainers {
+		if obj.Spec.InitContainers[i].Resources.Limits != nil {
```

Comment:
the PodSpec object is reused in lots of objects, the intent of this is to only apply defaulting to one inside a Pod object